base/shell/explorer/explorer.cpp

```diff
@@ -87,6 +87,43 @@ HideMinimizedWindows(IN BOOL bHide)
 }
 #endif
 
+static BOOL
+IsExplorerSystemShell()
+{
+    HKEY hKeyWinlogon;
+	WCHAR szShell[256];
+	WCHAR szPath[MAX_PATH];
+    LPWSTR szExplorer = NULL;
+	DWORD dwType;
+	DWORD dwBufferSize = sizeof(szShell);
+
+    if (!GetModuleFileName(NULL, szPath, MAX_PATH))
+        return FALSE;
+
+    szExplorer = PathFindFileName((LPWSTR)szPath);
+
+	if (RegOpenKeyEx(HKEY_LOCAL_MACHINE, L"Software\\Microsoft\\Windows NT\\CurrentVersion\\Winlogon",
+		0, KEY_READ, &hKeyWinlogon) != ERROR_SUCCESS)
+    {
+        // No registry access.
+        return TRUE;
+    }
+
+    if (RegQueryValueEx(hKeyWinlogon, L"Shell", 0, &dwType,
+		(LPBYTE)szShell, &dwBufferSize) == ERROR_SUCCESS)
+	{
+		RegCloseKey(hKeyWinlogon);
+
+		if (StrStrI(szShell, szExplorer))
+			return TRUE;
+		else
+			return FALSE;
+	}
+
+	// Unable to query value.
+	return FALSE;
```

Comment:
Unable to read the value (missing etc.) should return TRUE?

Reply:
I suppose. I just wasn't sure what the "failsafe" should be - should Explorer assume it is the shell and show the taskbar more often than not? or rather just be a file browser on the safe side? I will change this to TRUE though.